pkgs/development/compilers/opensmalltalk-vm/build-vm.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchurl
+, alsa-lib
+, coreutils
+, file
+, freetype
+, gnugrep
+, libpulseaudio
+, libtool
+, libuuid
+, openssl
+, pango
+, pkg-config
+, xorg
+,
+}:
+let
+  owner = "OpenSmalltalk";
+  repo = "opensmalltalk-vm";
+  version = "202206021410";
+in
+{ platformDir, vmName, configureFlagsArray, configureFlags }:
+stdenv.mkDerivation {
+  pname =
+    let vmNameNoDots = builtins.replaceStrings [ "." ] [ "-" ] vmName;
+    in "opensmalltalk-vm-${platformDir}-${vmNameNoDots}";
+  inherit version;
+
+  src = fetchFromGitHub {
+    inherit owner repo;
+    rev = version;
+    hash = "sha256-QqElPiJuqD5svFjWrLz1zL0Tf+pHxQ2fPvkVRn2lyBI=";
+  };
+
+  postPatch =
+    let
+      inherit (builtins) substring;
+      url = "https://github.com/${owner}/${repo}.git";
+      year = substring 0 4 version;
+      month = substring 4 2 version;
+      day = substring 6 2 version;
+      hour = substring 8 2 version;
+      minute = substring 10 2 version;
+      date = "${year}-${month}-${day}T${hour}:${minute}+0000";
+      commitHash = fetchurl {
+        url = "https://api.github.com/repos/${owner}/${repo}/commits/${version}";
+        curlOpts = "--header Accept:application/vnd.github.v3.sha";
+        hash = "sha256-jhj+bngESaWC69T/fz9rMSFIvcqn7h001QMShMDrnkc=";
+      };
```

Comment:
Can we just replace this with the version? This will likely run into rate limits for some people and thus fail.

Reply:
It seems from [`sqSCCSVersion.h`](https://github.com/OpenSmalltalk/opensmalltalk-vm/blob/202206021410/platforms/Cross/vm/sqSCCSVersion.h#L76-L87) that the git hash is meant to be exposed separately from the version tag, so I want to try and respect that. How about I hardcode the commit hash instead?

Reply:
We can just fill that with dummy values to make this a lot simpler but sure if you like to manually add that.